corehq/form_processor/models.py

```diff
@@ -47,6 +74,102 @@ def get(cls, id):
+    @property
+    def is_deprecated(self):
+        return self.state == self.DEPRECATED
+
+    @property
+    def is_duplicate(self):
+        return self.state == self.DUPLICATE
+
+    @property
+    def is_error(self):
+        return self.state == self.ERROR
+
+    @property
+    def is_submission_error_log(self):
+        return self.state == self.SUBMISSION_ERROR_LOG
+
+    @property
+    @memoized
+    def form_data(self):
+        from .utils import convert_xform_to_json
+        xml = self._get_xml()
+        return convert_xform_to_json(xml)
+
+    @property
+    def history(self):
+        return self.xformoperationsql_set.order_by('date')
+
+    @property
+    def metadata(self):
+        if const.TAG_META in self.form_data:
+            return XFormMetadata.wrap(self._clean_metadata(self.form_data[const.TAG_META]))
+
+        return None
+
+    def _clean_metadata(self, meta_block):
+        from .utils import get_text_attribute
+
+        if not meta_block:
+            return meta_block
+        meta_block = self._remove_unused_meta_attributes(meta_block)
+        meta_block['appVersion'] = get_text_attribute(meta_block.get('appVersion'))
+        meta_block['location'] = get_text_attribute(meta_block.get('location'))
+        meta_block = self._parse_meta_times(meta_block)
+
+        # also clean dicts on the return value, since those are not allowed
+        for key in meta_block:
+            if isinstance(meta_block[key], dict):
+                meta_block[key] = self._flatten_dict(meta_block[key])
+
+        return meta_block
+
+    def _flatten_dict(self, dictionary):
+        return ", ".join("{}:{}".format(k, v) for k, v in dictionary.items())
+
+    def _remove_unused_meta_attributes(self, meta_block):
+        for key in meta_block.keys():
+            # remove attributes from the meta block
+            if key.startswith('@'):
+                del meta_block[key]
+        return meta_block
+
+    def _parse_meta_times(self, meta_block):
+        for key in ("timeStart", "timeEnd"):
+            if meta_block.get(key, None):
+                if re_date.match(meta_block[key]):
+                    # this kind of leniency is pretty bad and making it midnight in UTC
+                    # is totally arbitrary here for backwards compatibility
+                    meta_block[key] += 'T00:00:00.000000Z'
+                try:
+                    # try to parse to ensure correctness
+                    parsed = iso_string_to_datetime(meta_block[key])
+                    # and set back in the right format in case it was a date, not a datetime
+                    meta_block[key] = json_format_datetime(parsed)
+                except Exception:
+                    logging.exception('Could not parse meta_block')
+                    # we couldn't parse it
+                    del meta_block[key]
+            else:
+                # it was empty, also a failure
+                del meta_block[key]
+
+        return meta_block
```

Comment:
agreed. was thinking about doing that but for some reason didn't go with it. will update